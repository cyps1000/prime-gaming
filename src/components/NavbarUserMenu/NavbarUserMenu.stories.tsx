/**
 * Storybook
 * @see https://storybook.js.org/docs/react/get-started/introduction
 */
import { Story, Meta } from "@storybook/react";

/**
 * Component Imports
 */
import NavbarUserMenu from "./NavbarUserMenu";

export default {
  title: "Components/NavbarUserMenu",
  component: NavbarUserMenu,
} as Meta;

/**
 * Defines the Template
 * @param args NavbarUserMenuProps
 * @returns
 */
const Template: Story = (args) => (
  <NavbarUserMenu
    openSignUpModal={() => {}}
    openSignInModal={() => {}}
    {...args}
  />
);

/**
 * Default case
 */
export const Default = Template.bind({});

/**
 * Put your component props in here
 *
 * Example:
 * Default.args = {
 *  text: "Hello World"
 * }
 *
 * Assuming that the component expects a prop text that is a string.
 */
Default.args = {};
