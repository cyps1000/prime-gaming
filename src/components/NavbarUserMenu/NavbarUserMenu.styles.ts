/**
 *  Material UI Imports
 * @see https://material-ui.com/styles/basics/
 */
import { makeStyles, fade } from "@material-ui/core/styles";

/**
 * Styles the component
 */
const useStyles = makeStyles((theme) => ({
  menuItems: {
    display: "flex",
    alignItems: "center",
    flex: 3,
    justifyContent: "flex-end",
    paddingRight: "2.5rem",
    "& button": {
      "&:hover": {
        backgroundColor: fade(theme.palette.common.white, 0.15),
      },
      margin: "0 10px",
      fontSize: 16,
      color: theme.palette.secondary.main,
      fontWeight: 800,
      textTransform: "initial",
    },
  },
}));

export { useStyles };
