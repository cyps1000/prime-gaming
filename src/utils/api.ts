import axios from "axios";

/**
 * Defines the hook props interface
 */
interface Config {
  mock: boolean;
}

export const getApiClient = (config: Config) => {
  const { mock } = config;

  /**
   * Handles getting the base api url
   */
  const getApiUrl = () => {
    if (process.env.NODE_ENV === "development") {
      if (mock) return "/";
      return process.env.REACT_APP_LOCAL_API;
    }
    return process.env.REACT_APP_STAGING_API;
  };

  const apiClient = axios.create({
    baseURL: getApiUrl(),
  });

  return { apiClient };
};
