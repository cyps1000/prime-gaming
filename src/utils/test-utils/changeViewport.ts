/**
 * External imports
 */
import mediaQuery from "css-mediaquery";

/**
 * Defines the BreakpointKey type
 */
type BreakpointKey = "xl" | "lg" | "md" | "sm" | "xs";

/**
 * Handles getting the width based on the breakpoint key provided
 */
const getWidthByThemeBreakpoints = (breakpointKey: BreakpointKey) => {
  switch (breakpointKey) {
    case "xl":
      return 1920;
    case "lg":
      return 1280;
    case "md":
      return 960;
    case "sm":
      return 600;
    case "xs":
      return 414;
    default:
      return 1920;
  }
};

/**
 * Handles changing the viewport
 */
export const changeViewport = (breakpointKey: BreakpointKey): any => {
  let width: number = getWidthByThemeBreakpoints(breakpointKey);

  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width: width }),
      addListener: () => {},
      removeListener: () => {},
    };
  };
};
