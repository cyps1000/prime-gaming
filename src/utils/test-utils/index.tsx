import React, { ReactElement } from "react";
/**
 * External Imports
 */
import { MemoryRouter } from "react-router-dom";

/**
 * Imports testing utils
 */
import { render, RenderOptions } from "@testing-library/react";

/**
 * Imports providers
 */

import { ThemeProvider, LanguageProvider } from "../../hooks";
import Providers from "../../components/Providers";

/**
 * Defines the Wrapper component
 */
const Wrapper: React.FC = ({ children }) => {
  return (
    <ThemeProvider>
      <LanguageProvider>
        <Providers>
          <MemoryRouter>{children}</MemoryRouter>
        </Providers>
      </LanguageProvider>
    </ThemeProvider>
  );
};

/**
 * Defines the custom render
 */
const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: Wrapper, ...options });

/**
 * Re-exports everything from the testing library
 */
export * from "@testing-library/react";

export * from "./changeViewport";
export * from "./mocks";
export { default as userEvent } from "@testing-library/user-event";
export { customRender as render };
