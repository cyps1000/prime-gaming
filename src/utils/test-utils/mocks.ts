export const mockHistoryPush = jest.fn();
