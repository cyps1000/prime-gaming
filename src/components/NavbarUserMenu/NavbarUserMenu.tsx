/**
 * Modules Imports
 */
import { useTranslation } from "react-i18next";
import { useHistory } from "react-router-dom";

/**
 * Imports Material UI Components
 */
import Button from "@material-ui/core/Button";

/**
 * Imports the component styles
 */
import { useStyles } from "./NavbarUserMenu.styles";

/**
 * Displays the component
 */
const NavbarUserMenu: React.FC = (props) => {
  /**
   * Handles the translations
   */
  const { t } = useTranslation();

  /**
   * Gets the component styles
   */
  const classes = useStyles();

  /**
   * Gets the history object
   */
  const history = useHistory();

  /**
   * Handles routing
   */
  const routeTo = (url: string) => {
    history.push(url);
  };

  /**
   * Defines the routing functions
   */
  const goToHome = () => routeTo("/");
  const goToNews = () => routeTo("/news");
  const goToAbout = () => routeTo("/about");
  const goToContact = () => routeTo("/contact");
  const goToRegister = () => routeTo("/register");

  return (
    <div className={classes.menuItems}>
      <Button onClick={goToHome}>{t("home")}</Button>
      <Button onClick={goToNews}>{t("news")}</Button>
      <Button onClick={goToAbout}>{t("about")}</Button>
      <Button onClick={goToContact}>{t("contact")}</Button>
      <Button>{t("signIn")}</Button>
      <Button onClick={goToRegister}>{t("signUp")}</Button>
    </div>
  );
};

export default NavbarUserMenu;
