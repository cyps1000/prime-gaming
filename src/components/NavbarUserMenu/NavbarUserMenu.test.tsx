/**
 * @see https://testing-library.com/docs/react-testing-library/intro
 * @see https://www.robinwieruch.de/react-testing-library
 * @see https://www.smashingmagazine.com/2020/07/react-apps-testing-library/
 */

/**
 * Imports test utils
 */
import { render, fireEvent } from "../../utils/test-utils";
import { mockHistoryPush } from "../../utils/test-utils/mocks";

/**
 * External Imports
 */
import pretty from "pretty";

/**
 * Imports component
 */
import NavbarUserMenu from "./NavbarUserMenu";

/**
 * NavbarUserMenu rendering tests
 */
describe("NavbarUserMenu Rendering Tests", () => {
  it("renders the component without errors", () => {
    render(<NavbarUserMenu />);
  });
});

/**
 * NavbarUserMenu logic tests
 */
describe("NavbarUserMenu Logic Tests", () => {
  it("takes the user to the home page when clicked", () => {
    const { getByText } = render(<NavbarUserMenu />);

    fireEvent.click(getByText(global.en_en["home"]));
    expect(mockHistoryPush).toHaveBeenCalledWith("/");

    fireEvent.click(getByText(global.en_en["news"]));
    expect(mockHistoryPush).toHaveBeenCalledWith("/news");

    fireEvent.click(getByText(global.en_en["about"]));
    expect(mockHistoryPush).toHaveBeenCalledWith("/about");

    fireEvent.click(getByText(global.en_en["contact"]));
    expect(mockHistoryPush).toHaveBeenCalledWith("/contact");

    expect(mockHistoryPush).toHaveBeenCalledTimes(4);
  });
});

/**
 * NavbarUserMenu snapshot test
 */
describe("NavbarUserMenu Snapshot Test", () => {
  it("passes the snapshot test", () => {
    const { container } = render(<NavbarUserMenu />);

    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
