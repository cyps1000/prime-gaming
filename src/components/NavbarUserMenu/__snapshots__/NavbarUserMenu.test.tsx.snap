// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarUserMenu Snapshot Test passes the snapshot test 1`] = `"<div class=\\"makeStyles-menuItems-3\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Dashboard</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Home</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">News</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">About</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Contact</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Sign In</span><span class=\\"MuiTouchRipple-root\\"></span></button><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">Sign Up</span><span class=\\"MuiTouchRipple-root\\"></span></button></div>"`;
